### PR TITLE
boards: stm32f746g_disco: Enable a few peripherals 

### PR DESCRIPTION
**Pull request by chrta**
This enables I2C, SPI, PWM, RTC and USB.

RTC tested on samples/drivers/rtc
USB tested on samples/subsys/usb/cdc_acm

Open issues:
1. I2C and SPI are not tested, due to the lack of hardware.

### CODE REVIEW

**Comment by chrta** [comment]
Pinmux defines for SPI are missing.

File: dts/arm/st/f7/stm32f7.dtsi [hunk 1]
```diff
@@ -241,6 +241,17 @@
 			label = "I2C_3";
 		};
 
+		spi2: spi@40003800 {
```
Comment (erwango) [inline dts/arm/st/f7/stm32f7.dtsi]: Please populate all SPI devices applicable STM32F746xg

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32f7/dts_fixup.h [hunk 2]
```diff
@@ -134,4 +134,10 @@
 #define CONFIG_PWM_STM32_14_DEV_NAME		ST_STM32_PWM_40002000_PWM_LABEL
 #define CONFIG_PWM_STM32_14_PRESCALER		ST_STM32_PWM_40002000_PWM_ST_PRESCALER
 
+#define CONFIG_RTC_0_BASE_ADDRESS               ST_STM32_RTC_40002800_BASE_ADDRESS
+#define CONFIG_RTC_0_IRQ_PRI                    ST_STM32_RTC_40002800_IRQ_0_PRIORITY
+#define CONFIG_RTC_0_IRQ                        ST_STM32_RTC_40002800_IRQ_0
+#define CONFIG_RTC_0_NAME                       ST_STM32_RTC_40002800_LABEL
+#define CONFIG_RTC_PRESCALER                    ST_STM32_RTC_40002800_PRESCALER
+
```
[the remark this answers is not in the document]
Comment (chrta) [reply soc/arm/st_stm32/stm32f7/dts_fixup.h]: done

File: dts/arm/st/f7/stm32f7.dtsi (hunk 1 above)
Comment (chrta) [reply dts/arm/st/f7/stm32f7.dtsi]: done

**Comment by erwango** [comment]
@chrta : I'll push a PR to fix stm32cube issue

**Comment by chrta** [comment]
Rebased

File: boards/arm/stm32f746g_disco/pinmux.c [hunk 3]
```diff
@@ -35,6 +38,20 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PG13, STM32F7_PINMUX_FUNC_PG13_ETH},
 	{STM32_PIN_PG14, STM32F7_PINMUX_FUNC_PG14_ETH},
 #endif /* CONFIG_ETH_STM32_HAL */
+#ifdef CONFIG_I2C_1
+	{STM32_PIN_PB8, STM32F7_PINMUX_FUNC_PB8_I2C1_SCL},
+	{STM32_PIN_PB9, STM32F7_PINMUX_FUNC_PB9_I2C1_SDA},
+#endif  /* CONFIG_I2C_1 */
+#ifdef CONFIG_SPI_2
+	{STM32_PIN_PI0,  STM32F7_PINMUX_FUNC_PI0_SPI2_NSS},
+	{STM32_PIN_PI1,  STM32F7_PINMUX_FUNC_PI1_SPI2_SCK},
```
Comment (erwango) [inline boards/arm/stm32f746g_disco/pinmux.c]: Is STM32F7_PINMUX_FUNC_PI1_SPI2_SCK declared in drivers/pinmux/stm32/pinmux_stm32f7.h?

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/stm32f746g_disco/pinmux.c (hunk 3 above)
Comment (chrta) [reply boards/arm/stm32f746g_disco/pinmux.c]: Yes, i added all SPI2 pinmux definitions to this file.

**Review by erwango — APPROVED** [review]

**Comment by galak** [comment]
@dbkinder can you review the docs.

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks.

**Pull request merged**